app/components/search_preview/search_preview.js

```diff
@@ -52,54 +57,68 @@ export default class SearchPreview extends PureComponent {
     };
 
     state = {
-        showPosts: false,
-        animationEnded: false
+        show: false
     };
 
-    componentWillReceiveProps(nextProps) {
-        const {animationEnded, showPosts} = this.state;
-        if (animationEnded && !showPosts && nextProps.postIds.length) {
-            this.setState({showPosts: true});
+    componentWillMount() {
+        const {postIds} = this.props;
+        if (postIds && postIds.length >= 10) {
+            this.setState({show: true});
+        } else {
+            this.loadPosts();
```

Comment:
This feels odd to me, like the state could be set in the constructor and the check for `postIds` and `loadPosts` could be called in `componentDidMount`.